### PR TITLE
Refactor/update settings page

### PR DESCRIPTION
**Pull request by Keitarofreeman**
Related issues:
https://github.com/tabetalt/backoffice/issues/121
https://github.com/tabetalt/backoffice/issues/120
https://github.com/tabetalt/backoffice/issues/104

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,29 +1,32 @@
...
-  },
-  "settings": {
-    "react": {
-      "pragma": "React",
-      "version": "detect"
-    }
-  },
-  "parser": "@typescript-eslint/parser"
-}
+{
```
Comment (cobraz) [inline .eslintrc.json:1]: What is happening in this? 🤷‍♂️

File: codegen.yml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 overwrite: true
-schema: 'https://dev.v1.api.tabetalt.no/'
+schema: http://localhost:3000/
```
Comment (cobraz) [inline codegen.yml]: Under normal conditions, we probably want to use the external API.

```suggestion
schema: https://dev.v1.api.tabetalt.no/
```

What do you think?

File: package.json [hunk 3]
```diff
@@ -1,84 +1,84 @@
...
-    "rimraf": "^3.0.2"
-  },
-  "husky": {
-    "hooks": {
-      "pre-commit": "yarn lint && yarn test:ci"
-    }
-  }
-}
\ No newline at end of file
+{
```
Comment (cobraz) [inline package.json:1]: Here too? What's happening?

File: src/App.tsx [hunk 4]
```diff
@@ -6,6 +6,7 @@ import { theme, Skeleton } from '@tabetalt/kit';
 import { getRoutes } from './routing';
 import gqlClient from './api/client';
 import { useAuth } from './context/AuthContext';
+import { TenantsProvider } from './context/TenantsContext';
```
Comment (cobraz) [inline src/App.tsx:9]: We probably want to use singular naming `tenants` → `tenant`

File: src/context/TenantsContext.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  currentTenant: TenantItem | null;
+  updateRequired: () => void;
+}
+
+export const TenantsContext = React.createContext<TenantContextValue>({
+  tenants: [],
+  currentTenant: null,
+  updateRequired: () => {
+    // Dummy function
+  },
```
Comment (cobraz) [inline src/context/TenantsContext.tsx:18]: What the intention with this?

File: src/context/TenantsContext.tsx [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+function useAuthContext() {
+  const context = useAuth();
+  return context;
+}
+
+export const TenantsProvider: React.FC = ({ children }) => {
+  const authContext = useAuthContext();
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [currentUserId, setUserId] = useState<string>();
```
Comment (cobraz) [inline src/context/TenantsContext.tsx:33]:
```suggestion
  const [, setUserId] = useState<string>();
```

File: src/helpers/index.ts [hunk 7]
```diff
@@ -15,6 +14,15 @@ export const formatPrice = (
   return amount;
 };
 
+export const formatMoney = (value?: Money | null): string => {
+  if (!value) return '0';
+
+  const priceValue = value as DineroObject;
+
+  const amount = priceValue ? Dinero(priceValue).toFormat('0.00') : '0';
```
Comment (cobraz) [inline src/helpers/index.ts:22]:
```suggestion
  const amount = value as DineroObject ? Dinero(priceValue).toFormat('0.00') : '0';
```

Type-casting isn't super pretty. We should probably have some util that converts `Money` to `DineroObject`, but this is fine for now :)

File: src/helpers/index.ts [hunk 8]
```diff
@@ -40,3 +48,23 @@ export const moneyFromString = (
 
   return price;
 };
+
+export const vatRateCalculation = (
+  vatRateObject: DineroObject,
+  price: string
+): string => {
+  const dineroPrice = Dinero(moneyFromString(price) as DineroObject);
```
Comment (cobraz) [inline src/helpers/index.ts:56]: This looks a bit _wordy_. Maybe we should make a function like mentioned above? `dineroFromMoney` or something? 🤷‍♂️

File: src/modules/catalog/product/ProductCreate.tsx [hunk 9]
```diff
@@ -37,7 +37,6 @@ const ProductCreate: React.FC = () => {
       };
 
       if (values.categories) {
-        console.log(values.categories);
```
Comment (cobraz) [inline src/modules/catalog/product/ProductCreate.tsx:40]: Nice catch!

File: src/modules/catalog/products/Products.tsx [hunk 10]
```diff
@@ -34,9 +34,10 @@ const Products: React.FC = () => {
   } else if (error) {
     content = <span>Error getting product list</span>; // TODO: improve error message
   } else {
+    console.log('qweqweqeqweqw');
+    console.log(data);
```
Comment (cobraz) [inline src/modules/catalog/products/Products.tsx:38]:
```suggestion
```

**Review by cobraz — COMMENTED** [review]
Consider using less type-casting. I think we should try to remove the amount of `as XXXX` to use a function, this way we have one place to do type-casting.

**Comment by cobraz** [comment]
> This pull request **introduces 6 alerts** when merging [6128ce3](https://github.com/tabetalt/backoffice/commit/6128ce3f396e533db46962ed9bd6313589d65a76) into [d161548](https://github.com/tabetalt/backoffice/commit/d16154898bf662d28fc4f5e002a1e8e48dbca322) - [view on LGTM.com](https://lgtm.com/projects/g/tabetalt/backoffice/rev/pr-79c0aa842ff1acd43bde2309961a425dd6c10708)
> 
> **new alerts:**
> 
> * 6 for Unused variable, import, function or class

Should probably fix these before merging too. 👍

File: package.json (hunk 3 above)
Comment (Keitarofreeman) [reply package.json]: Sometimes vs code try to convert LF line endings to CRLF. I fixed it.

File: codegen.yml (hunk 2 above)
Comment (Keitarofreeman) [reply codegen.yml]: Fixed

**Review by cobraz — APPROVED** [review]
👍

**Pull request merged**